BAZProject/HomeMovieViper/View/HomeViewCollectionsExtension.swift

```diff
@@ -0,0 +1,78 @@
+//
+//  HomeViewCategoriesCollectionExtension.swift
+//  BAZProject
+//
+//  Created by 1050210 on 31/01/23.
+//
+
+import UIKit
+
+// MARK: - CollectionView's DataSource
+
+extension HomeMoviesView: UICollectionViewDataSource{
+    func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
+        if collectionView == self.categoriesMoviesCollectionView{
+            return presenter?.getCategoriesMoviesCount() ?? 0
+        }else{
+            return presenter?.getMoviesCount() ?? 0
+        }
+    }
+    
+    func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
+        if collectionView == self.categoriesMoviesCollectionView{
+            guard let cell = collectionView.dequeueReusableCell(withReuseIdentifier: "CategoriesMoviesCollectionViewCell", for: indexPath) as? CategoriesMoviesCollectionViewCell
+            else { return UICollectionViewCell() }
+            
+            presenter?.getCategorieImage(index: indexPath.row, completion: { categoryImage in
+                cell.setupCategoryImage(for: categoryImage ?? UIImage())
+            })
+            cell.delegate = self
+            cell.setupCell(indexPath: indexPath.row)
+            
+            return cell
+        }else{
+            guard let cell = collectionView.dequeueReusableCell(withReuseIdentifier: "MovieCollectionViewCell", for: indexPath) as? MovieCollectionViewCell else { return UICollectionViewCell() }
+            
+            presenter?.getImage(index: indexPath.row, completion: { imageMovie in
+                cell.setupMovieImage(for: imageMovie ?? UIImage())
```

Comment:
Maybe it would be better to send a nil when there's no image and the cell could set a default image for offline mode